specification/signalr/resource-manager/Microsoft.SignalR/preview/2018-03-01-preview/signalr.json

```diff
@@ -0,0 +1,1034 @@
+    },
+    "host": "management.azure.com",
+    "schemes": [
+        "https"
+    ],
+    "consumes": [
+        "application/json"
+    ],
+    "produces": [
+        "application/json"
+    ],
+    "paths": {
+        "/providers/Microsoft.SignalR/operations": {
+            "get": {
+                "tags": [
+                    "SignalR"
+                ],
+                "description": "Lists all of the available REST API operations of the Microsoft.SignalR provider.",
+                "operationId": "SignalR_ListOperations",
+                "parameters": [
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "Success. The response describes the list of operations.",
+                        "schema": {
+                            "$ref": "#/definitions/OperationListResponse"
+                        }
+                    }
+                },
+                "x-ms-examples": {
+                    "ListOperations": {
+                        "$ref": "./examples/ListOperations.json"
+                    }
+                }
+            }
+        },
+        "/subscriptions/{subscriptionId}/providers/Microsoft.SignalR/checkNameAvailability": {
+            "post": {
+                "tags": [
+                    "SignalR"
+                ],
+                "description": "Checks that the SignalR name is valid and is not already in use.",
+                "operationId": "SignalR_CheckNameAvailability",
+                "parameters": [
+                    {
+                        "name": "nameAvailabilityRequest",
+                        "in": "body",
+                        "description": "parameters supplied to the operation.",
+                        "required": false,
+                        "schema": {
+                            "$ref": "#/definitions/NameAvailabilityRequest"
+                        }
+                    },
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/SubscriptionIdParameter"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "Success. The response describes the name availability.",
+                        "schema": {
+                            "$ref": "#/definitions/CheckNameAvailabilityResponse"
+                        }
+                    }
+                },
+                "x-ms-examples": {
+                    "CheckNameAvailability": {
+                        "$ref": "./examples/CheckNameAvailability.json"
+                    }
+                }
+            }
+        },
+        "/subscriptions/{subscriptionId}/providers/Microsoft.SignalR/SignalR": {
```

Comment:
Will this result be paged, or will you always return the entire list of resources?